errors.go

```diff
@@ -0,0 +1,248 @@
+package gobayeux
+
+import (
+	"errors"
+	"fmt"
+)
+
+var (
```

Comment:
These should be constants

Reply:
Sadly `errors.New` is [not constant](https://go.dev/play/p/WM6p6JndA8N).

Reply:
So right now, someone could mistakenly replace this, e.g., 

```go
func init() {
    gobayeux.ErrClientNotConnected = errors.New("something else")
}
```

If we did something like

```go
type constError string

func (c constError) Error() string {
    return string(c)
}

var _ error = (*constError)(nil)

const (
ErrClientNotConnected = constError("client not connected to server")
)
````

Reply:
I don't think you need the `var _ error = (*constError)(nil)` line do you?

Reply:
Looks like you don't. :shrug:

Reply:
It makes me wonder why the `errors` package doesn't provide a constant sentinel error type.